Topics/Tech_Stacks/Shadcn.md

```diff
@@ -0,0 +1,144 @@
+           destructive:
+           "bg-destructive text-destructive-foreground hover:bg-destructive/90",
+           outline:
+           "border border-input bg-background hover:bg-accent hover:text-accent-foreground",
+           secondary:
+           "bg-secondary text-secondary-foreground hover:bg-secondary/80",
+           ghost: "hover:bg-accent hover:text-accent-foreground",
+           link: "text-primary underline-offset-4 hover:underline",
+       },
+       size: {
+           default: "h-10 px-4 py-2",
+           sm: "h-9 rounded-md px-3",
+           lg: "h-11 rounded-md px-8",
+           icon: "h-10 w-10",
+       },
+       },
+       defaultVariants: {
+       variant: "default",
+       size: "default",
+       },
+   }
+   )
+
+   export interface ButtonProps
+   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+       VariantProps<typeof buttonVariants> {
+   asChild?: boolean
+   }
+
+   const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
+   ({ className, variant, size, asChild = false, ...props }, ref) => {
+       const Comp = asChild ? Slot : "button"
+       return (
+       <Comp
+           className={cn(buttonVariants({ variant, size, className }))}
+           ref={ref}
+           {...props}
+       />
+       )
+   }
+   )
+   Button.displayName = "Button"
+
+   export { Button, buttonVariants }
+
+   ```
+
+2. **Usage:** Import the `Button` component in your project wherever you need to use it.
+
+   ```Typescript
+   import { Button } from "@/components/ui/button"
+   ```
+
+3. **Customization:** Add the imported component in your project and customize it however you like. See [examples](https://ui.shadcn.com/docs/components/button#examples) of how to use the different variants defined in [button.tsx](#buttontsx) to style your component in different ways.
+
+   #### Customizing Using Variants
+
+   The variants given to you in the file is named according to the colours that are defined in your main css file during installation.
+
+   You can make changes to the default variants with Tailwind to fit your project's theme.
+
+   You can also rename the variants and add your own variants to create more customization in your component.
+
+   To apply a certain variant to your component, all you need to do is define the variant property to your used component:
+
+   ```Typescript
+   <Button variant="outline">Click here</Button>
+   ```
+
+   #### Customizing the Size
+
+   Sizing of the component can be customized to easily create responsive designs. Similar to variants, you can customize different sizes in the component file and use them in your project:
+
+   ```Typescript
+   <Button variant="outline" size="lg">Click here</Button>
+   ```
+
+   Similar to variants, you can rename the default sizes or add your own.
+
+See the shadcn/ui [documentation](https://ui.shadcn.com/docs/components/accordion) to see all the components that are available to re-use in your project.
```

Comment:
@neecofabian @sarahhjchung This is an open-ended thought, but what are your thoughts on adding other visual references to this article? Since ShadCN is a visual component library I'm thinking that it'll be relevant. Maybe images of the different buttons as you're progressing through the example? Feel free to discuss!

Reply:
That's a great idea, I think it can bring some visual differences to this new page and show off the library's components. In addition to showing the button's progress throughout the example, @sarahhjchung can include the different button variants to show how CVA variant properties affect shadcn objects

Reply:
After addressing all the other comments you and Neeco have made, I'll see if there's any good spots to add some visual references in. I think it might be helpful to add screenshots of the different button variants like you're suggesting.

Reply:
After address the comments, I thought it would be best to share a link to view the button variant examples in the shadcn documentation. Personally, it looks cleaner and focuses the attention on the code snippets that I included.

Reply:
@sarahhjchung that makes sense! Just took a look and I really like the addition of this resources link. As I mentioned in a previous comment, readers who are interested in exploring ShadCN will naturally follow this path to investigating the component library for themselves.